src/lib/perf/perf_counter.cpp

```diff
@@ -188,7 +184,7 @@ perf_alloc_once(enum perf_counter_type type, const char *name)
 void
 perf_free(perf_counter_t handle)
 {
-	if (handle == NULL) {
+	if (handle == nullptr) {
```

Comment:
Is `delete` happening in `sq_rem()` just below?

Reply:
Whoops, got so side tracked adding all the cmake exceptions that I didn't even finish the original purpose of the PR.

I'd really like to evolve perf counters into a version we can embed in module classes rather than dynamically allocate. I want them to auto include the MODULE_NAME and handle multiple instances.

Reply:
Delete is now happening in perf_free().

Reply:
Aswesome!